tests/AggregateRoots/AggregateEntityTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Spatie\EventSourcing\Tests\AggregateRoots;
+
+use Exception;
+use Spatie\EventSourcing\AggregateRoots\AggregateEntity;
+use Spatie\EventSourcing\AggregateRoots\AggregateRoot;
+use Spatie\EventSourcing\StoredEvents\Models\EloquentStoredEvent;
+use Spatie\EventSourcing\StoredEvents\ShouldBeStored;
+use Spatie\EventSourcing\Tests\TestCase;
+
+class AggregateEntityTest extends TestCase
+{
+    private const CART_UUID = 'cart-uuid';
+
+    /** @test */
+    public function test_entities()
+    {
+        $cart = Cart::retrieve(self::CART_UUID);
+
+        $cart
+            ->addItem('test')
+            ->addItem('test 2')
+            ->persist();
+
+        $this->assertDatabaseCount((new EloquentStoredEvent)->getTable(), 2);
+
+        $cart::retrieve(self::CART_UUID);
+
+        $this->assertCount(2, $cart->cartItems->items);
+        $this->assertEquals('test', $cart->cartItems->items[0]);
+        $this->assertEquals('test 2', $cart->cartItems->items[1]);
+
+        $cart->clear()->persist();
+
+        $this->assertExceptionThrown(function () use ($cart) {
+            $cart->checkout();
+        });
+
+        $cart::retrieve(self::CART_UUID);
+
+        $this->assertExceptionThrown(function () use ($cart) {
+            $cart->checkout();
+        });
+    }
+}
+
+class Cart extends AggregateRoot
```

Comment:
Could you move these test classes each in their own files in a `TestClasses` namespace? That seems cleaner to me.